src/services/spectral-agent-service.ts

```diff
@@ -96,12 +96,19 @@ export class SpectralAgentService {
   }
 
   public processResults(results: ScanResult, folderPath: string) {
-    this.resetFindings()
     results?.items?.forEach((item: ScanFindingView) => {
       this.processFindingItem({ item, folderPath: folderPath.toLowerCase() })
     })
   }
 
+  public resetFindings() {
+    this.findings = { [FindingType.iac]: {}, [FindingType.secret]: {} }
```

Comment:
in general in this codebase we should try to avoid explicit finding types, should be generic.
for example, this code doesn't support oss type, and any other type we'll add in the future we'll have to remember to add it here.
I think we can build the map from the results, or at minimum keep the types in a single place and take it from there.

Reply:
Record is made for these cases like I commented here
https://github.com/SpectralOps/vscode-extension/pull/8#discussion_r1233224697

Reply:
fixed https://github.com/SpectralOps/vscode-extension/pull/8/commits/fffbca1c6489b308e5c02704e1ce97ef8d208ff4